src/main/java/mate/academy/onlinebookstore/service/order/OrderSeviceImpl.java

```diff
@@ -0,0 +1,95 @@
+import lombok.RequiredArgsConstructor;
+import mate.academy.onlinebookstore.dto.order.CreateOrderDto;
+import mate.academy.onlinebookstore.dto.order.OrderDto;
+import mate.academy.onlinebookstore.dto.order.UpdateStatusOrderDto;
+import mate.academy.onlinebookstore.mapper.OrderMapper;
+import mate.academy.onlinebookstore.model.CartItem;
+import mate.academy.onlinebookstore.model.Order;
+import mate.academy.onlinebookstore.model.OrderItem;
+import mate.academy.onlinebookstore.model.ShoppingCart;
+import mate.academy.onlinebookstore.model.User;
+import mate.academy.onlinebookstore.repository.order.OrderRepository;
+import mate.academy.onlinebookstore.repository.shoppingcart.ShoppingCartRepository;
+import mate.academy.onlinebookstore.repository.user.UserRepository;
+import mate.academy.onlinebookstore.service.cartitem.CartItemService;
+import mate.academy.onlinebookstore.service.orderitem.OrderItemService;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderSeviceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final OrderMapper orderMapper;
+    private final UserRepository userRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemService cartItemService;
+    private final OrderItemService orderItemService;
+    
+    @Override
+    public OrderDto save(String userEmail, CreateOrderDto createOrderDto) {
+        User currentUser = findUserByEmail(userEmail);
+        ShoppingCart userShoppingCart = shoppingCartRepository.findByUserId(currentUser.getId());
+        List<CartItem> existedCartItemsList = cartItemService
+                .findByShoppingCartId(userShoppingCart.getId());
+
+        if (existedCartItemsList.size() == 0) {
+            throw new UnsupportedOperationException("Can't create new order without "
+                    + "any items in the shopping cart");
+        }
+
+        Order order = new Order();
+        order.setUser(currentUser);
+        order.setStatus(Order.Status.PENDING);
+        order.setTotal(getTotalItemsPrice(existedCartItemsList));
+        order.setOrderDate(LocalDateTime.now());
+        order.setShippingAddress(createOrderDto.shippingAddress());
+
+        orderRepository.save(order);
+        order.setOrderItems(getOrderItems(order, existedCartItemsList));
+        return orderMapper.toDto(order);
+    }
+
+    @Override
+    public List<OrderDto> getAll(String userEmail) {
+        return orderRepository.findAll().stream()
+                .map(orderMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public OrderDto updateStatus(Long id, UpdateStatusOrderDto updateStatusOrderDto) {
+        Order orderById = orderRepository.findById(id).orElseThrow(() ->
+                new NoSuchElementException("Can't find the order bi id " + id));
+        orderById.setStatus(Order.Status.valueOf(updateStatusOrderDto.status()));
```

Comment:
Why dto has String status and without validation?
You will get here exception in case received inexisting status but have to manage it during dto validation.
Make dto field enum and ensure it's properly serialized if input correct or proper response provided in case input is invalid